lib/usb/usb_standard.c

```diff
@@ -247,19 +248,30 @@ static int usb_standard_set_configuration(usbd_device *usbd_dev,
 					  struct usb_setup_data *req,
 					  uint8_t **buf, uint16_t *len)
 {
-	int i;
+	uint8_t i;
+	const struct usb_config_descriptor *cfg;
 
 	(void)req;
 	(void)buf;
 	(void)len;
 
-	/* Is this correct, or should we reset alternate settings. */
-	if (req->wValue == usbd_dev->current_config) {
-		return 1;
+	if (req->wValue > usbd_dev->desc->bNumConfigurations) {
+		return USBD_REQ_NOTSUPP;
 	}
 
 	usbd_dev->current_config = req->wValue;
 
+	if (usbd_dev->current_config) {
```

Comment:
So this is resetting the altsetting for all interfaces in this what, the current config?  Why does it need to do that, if we're going to be switching to a new config anyway?